pkg/route/validation/validation_test.go

```diff
@@ -1946,13 +1990,293 @@ func TestValidateTLS(t *testing.T) {
 					},
 				},
 			},
+
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid Reencrypt route as externalCertificate and certificate set",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationReencrypt,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "dummy",
+					},
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid Passthrough route as externalCertificate and certificate set",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationPassthrough,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "dummy",
+					},
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 2,
+		},
+		{
+			name: "Invalid Edge route as externalCertificate and certificate set",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "dummy",
+					},
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
 			expectedErrors: 1,
 		},
+		{
+			name: "Invalid Passthrough route as externalCertificate",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationPassthrough,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			opts:           routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			expectedErrors: 1,
+		},
+		{
+			name: "Invalid Edge route for externalCertificate as not authorized",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						Termination: routev1.TLSTerminationEdge,
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tls-secret",
+					Namespace: "sandbox",
+				},
+				Type: corev1.SecretTypeTLS,
+			},
```

Comment:
`secret` field retrival will happen after SAR verification, true,  but the function will not return after SAR verification, it will continue and will add another error in the list: `secret not found`. 
So, in total 6 errors will be returned, which is not we are expecting in this test case of unauthorized access.

Reply:
Right, I forgot the validation goes on. Fine to close this one.